packages/editor/src/editor-ui/welcome-modal/use-welcome-modal.ts

```diff
@@ -0,0 +1,36 @@
+import { fold } from 'fp-ts/lib/Either'
+import { pipe } from 'fp-ts/lib/function'
+import * as t from 'io-ts'
+import { useEffect, useState } from 'react'
+
+const localStorageKey = 'hasUserSeenWelcomeModal'
+
+export const useWelcomeModal = () => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  useEffect(() => {
+    const localStorageValue = localStorage.getItem(localStorageKey)
+    if (localStorageValue === null) return setIsOpen(true)
+    const hasSeenModal = decodeWelcomeModalData(localStorageValue)
```

Comment:
If you don't reuse the preferences:
Could we simplify this by just checking if the key exists?
That way we wouldn't need any type checks and parsing.